src/main/resources/project/procedures/preamble.groovy

```diff
@@ -74,60 +73,75 @@ public class ElectricCommander {
         def resp
         def credential
 
-        if('$[config]'.startsWith("/")){
+        if ('$[config]'.startsWith("/")) {
             def plugConf = PerformHTTPRequest(RequestMethod.GET, '/rest/v1.0$[config]', [])
             def credName = plugConf.getData().pluginConfiguration.credentialMappings.parameterDetail[0].parameterValue
             resp = ef.getFullCredential(jobStepId: jobStepId, credentialName: credName)
             credential = resp.credential
-        }else{
+            println("config resp data in if" + plugConf.getData())
+        } else {
             resp = PerformHTTPRequest(RequestMethod.GET, '/rest/v1.0/jobsSteps/' + jobStepId + '/credentials/$[config]', [])
             credential = resp.getData().credential
+            println("config resp data in else" + resp.getData())
         }
 
-        if( resp == null ) {
+        if (resp == null) {
             throw new Exception("Error : Invalid configuration $[config].");
         }
-        if(!credential && resp.status != 200) {
+        if (!credential && resp.status != 200) {
             throw new Exception("Commander did not respond with 200 for credentials")
         }
 
         userName = credential.userName
         password = credential.password
+        try {
+            serviceUrl = getCommanderProperty('service_url')
+            println "service_url: $serviceUrl"
+        } catch (Throwable e) {
+            println "service_url is not found (service_url field does not exist?),setting to default"
+            serviceUrl = "https://s3.amazonaws.com"
+        }
```

Comment:
Move this logic to separate method and use the return value here.

Reply:
@cmarunachalam  moved the logic to a new method. And removed the print statements